### PR TITLE
Added EV1527 contact sensor protocol.

### PR DESCRIPTION
**Pull request by mel0en**
Added EV1527 protocol. Tested with 2 sensors (door / motion) and both are working. Please double-check because it has been a while :)

### CODE REVIEW

File: libs/protocols/ev1527.c [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+	ev1527->parseBinary=&ev1527ParseBinary;
+}
+
+#ifdef MODULE
+void compatibility(struct module_t *module) {
+	module->name = "ev1527";
+	module->version = "1.0";
+	module->reqversion = "6.0";
+	module->reqcommit = "99";
```
Comment (CurlyMoo) [inline libs/protocols/ev1527.c]: Why 99?

File: libs/protocols/ev1527.c [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+	options_add(&ev1527->options, 'f', "closed", OPTION_NO_VALUE, DEVICES_STATE, JSON_STRING, NULL, NULL);
+	options_add(&ev1527->options, 0, "readonly", OPTION_HAS_VALUE, GUI_SETTING, JSON_NUMBER, (void *)0, "^[10]{1}$");
+
+	ev1527->parseBinary=&ev1527ParseBinary;
+}
+
+#ifdef MODULE
+void compatibility(struct module_t *module) {
+	module->name = "ev1527";
+	module->version = "1.0";
```
Comment (CurlyMoo) [inline libs/protocols/ev1527.c]: I would prefer a 0.x until we got confirmation from others.

File: libs/protocols/ev1527.c (hunk 1 above)
Comment (mel0en) [inline libs/protocols/ev1527.c]: No clue to be honest.

File: libs/protocols/ev1527.c (hunk 2 above)
Comment (mel0en) [inline libs/protocols/ev1527.c]: Changed.

File: libs/protocols/ev1527.c [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+	ev1527->parseBinary=&ev1527ParseBinary;
+}
+
+#ifdef MODULE
+void compatibility(struct module_t *module) {
+	module->name = "ev1527";
+	module->version = "0.3";
+	module->reqversion = "6.0";
+	module->reqcommit = "0";
```
Comment (CurlyMoo) [inline libs/protocols/ev1527.c]: This must be `NULL`

Comment (mel0en) [inline libs/protocols/ev1527.c]: Thanks. fixed.

Comment (mel0en) [inline libs/protocols/ev1527.c]: Can I start working on a wiki page or do you have some thoughts on this protocol?

**Comment by CurlyMoo** [comment]
Can you make sure you create separate branches for each protocol. I already see the elro and ev1527 mixing up.

**Comment by mel0en** [comment]
Sounds logical, will look at it.

**Comment by mel0en** [comment]
Will start working with branches.

**Comment by CurlyMoo** [comment]
Thanks. I will write a contribution tutorial tomorrow.

**Pull request closed without merging**